src/internal/testutil/auth.go

```diff
@@ -144,16 +144,16 @@ func CommitCnt(t *testing.T, c *client.APIClient, repo *pfs.Repo) int {
 }
 
 // PipelineNames returns the names of all pipelines that 'c' gets from
-// ListPipeline.
-//
-// TODO(CORE-1100): replace with version which knows about projects.
-func PipelineNames(t *testing.T, c *client.APIClient) []string {
+// ListPipeline in the specified project. "" project name means get all pipelines.
+func PipelineNames(t *testing.T, c *client.APIClient, project string) []string {
 	t.Helper()
 	ps, err := c.ListPipeline(false)
 	require.NoError(t, err)
 	result := make([]string, len(ps))
 	for i, p := range ps {
-		result[i] = p.Pipeline.Name
+		if project == "" || project == p.Pipeline.Project.Name {
+			result[i] = p.Pipeline.Name
```

Comment:
I am not convinced that this algorithm is correct.  If only a few pipelines survive the filtering, then random elements are equal to `""`.  Prefer:

```go
var result []string
for _, p := range ps {
  if project == ... {
      result = append(result, p.GetPipeline().GetName())
  }
}
return result
```

Reply:
good catch, fixed.